### ISSUE TITLE
SMA producing NaNs

### ISSUE THREAD

**Issue by Jark5455** [TA-Lib/ta-lib-python#601]
Hello, it appears that with SMA, even if only a few input values are NaNs, ta-libs entire output comes out as NaN. I don't believe  this is intended behavior, because even where there is a valid series of values where the SMA can be calculated, NaN values are being returned.

**Comment by mrjbq7** [comment]
Intended behavior that’s how the underlying library works. Maybe not desired but many libraries like pandas provide a fillna() method that you can determine how to deal with NaNs in your data set. On Jun 27, 2023, at 2:37 PM, Jark ***@***.***> wrote:﻿
Hello, it appears that with SMA, even if only a few input values are NaNs, ta-libs entire output comes out as NaN. I don't believe  this is intended behavior, because even where there is a valid series of values where the SMA can be calculated, NaN values are being returned.

**Comment by Jark5455** [comment]
Is it possible to have TA-Lib just ignore NaNs in calculations? For example, if a period was `[Nan, 0, 1, 2, 3]` it would just remove Nan from the calculation and return 6 / 5. Or is there some sort of SIMD / Vectorization limitation that is preventing this?

**Comment by mario4tier** [comment]
TL;DR At TA-Lib Core level, there is no plan to handle NaN in its input (neither +/- infinity).

Why?
(1) It is speed inefficient to check for NaN for every input element, but worst: it adds complexity to handle "NaN section(s)" in intermediate calculations.

(2) Each function will have to define a bunch of algo rules. As an example, what to do when the whole SMA period is NaN? ... output NaN? SMA is an easy one, it would get complicated and inconsistent quickly with other functions.

(3) About replacing NaN with an arbitrary number: NaN are ok as leading/trailing filler in a buffer (e.g. as done by some wrapper on top of TA-Lib), but blindly replacing NaN in the middle of the data feels to me like hiding a problem (a NaN should raise the question as to why it is there in the first place).
